### PR TITLE
* updated documentation for tradematching and comparison

### PR DESCRIPTION
**Pull request by kevindeforth**

### CODE REVIEW

File: source-xor/index.html.md [hunk 1]
```diff
@@ -330,6 +331,143 @@ Known mathematically as ‘Private Set Intersection’, this function computes t
...
+
+### Parameters:
+
+| parameter | type | default | value-range | optional |
+|---|---|---|---|---|
+| reveal | bool | none | true, false | mandatory |
+| _strategy | string | "auto" | "dc", "naive" | optional |
+
+* ```reveal``` a bool indicating if the algorithm should reveal all the matches (`true`) or just the sum of the matches (`false`)
+* ```_strategy``` defaults to "auto", which will choose the strategy. The strategy indicates the comparison algorithm to execute - `dc` (divide and conquer) or `naive`. As a general rule of thumb, `dc` performs better than `naive` if the inputs do not exceed 5 million rows, because it requires less communication rounds at the cost of increased memory and network size.
```
Comment (shiftux) [inline source-xor/index.html.md]: let's add: "precision or correctness is however guaranteed independent of the choice of algorithm. "

File: source-xor/index.html.md [hunk 2]
```diff
@@ -330,6 +331,143 @@ Known mathematically as ‘Private Set Intersection’, this function computes t
...
+| 0 |
+| 0 |
+| 0 |
+| 0 |
+| 10000 |
+| 0 |
+| 900000 |
+
+**Important note on the inputs:**
+In order to not reveal the magnitude of the largest position held by any player, both players should add a line at the end of their notional-inputs with value `2^52` in case of `reveal=true`, or value `2^28` in case of `reveal=false`. Those values correspond to the maximum value allowing for integer precision. In case of double-precision, the notional may range up to `2^52` in no-reveal mode as well.
```
Comment (shiftux) [inline source-xor/index.html.md]: @alpetric FYI max ghostline values

File: source-xor/index.html.md [hunk 3]
```diff
@@ -330,6 +331,143 @@ Known mathematically as ‘Private Set Intersection’, this function computes t
...
+The input-files may contain up to 10 million + 1 rows, however the engine was tested for inputs up to 15 million rows.
```
Comment (shiftux) [inline source-xor/index.html.md]: why say that?

**Comment by shiftux** [comment]
thanks for the doc! - This is great!
only thing i would see missing would be an example of an axeList input (you give examples for columns, but no grouped input example).

**Comment by shiftux** [comment]
until we expose the function on UI or API however, i would wait with publishing that, if OK for you

File: source-xor/index.html.md (hunk 3 above)
Comment (kevindeforth) [reply source-xor/index.html.md]: removed it :+1:

File: source-xor/index.html.md (hunk 1 above)
Comment (kevindeforth) [reply source-xor/index.html.md]: did so :+1:

**Comment by kevindeforth** [comment]
thanks for the comments, have a look at it, I think I improved the examples

**Comment by kevindeforth** [comment]
and sure thing, I consider this your repo - merge at your discretion

**Comment by kevindeforth** [comment]
closing this --> documentation can be found in the engine-wiki and can be included in no time with the help of @ngama75's perl script

**Pull request closed without merging**